feat(LLMO-1023): integrate logWrapper for trace ID logging

HollywoodTonight opened the pull request:
- Add logWrapper to wrapper chain
- Replace context.log with context.contextualLog throughout
- Update CoralogixLogger to use contextualLog
- Logs will now include AWS X-Ray trace IDs automatically

Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues


Thanks for contributing!

HollywoodTonight committed:
feat(LLMO-1023): integrate logWrapper for trace ID logging

- Add logWrapper to wrapper chain
- Replace context.log with context.contextualLog throughout
- Update CoralogixLogger to use contextualLog
- Logs will now include AWS X-Ray trace IDs automatically

iuliag reviewed (commented):
🤔 I wasn't expecting that we needed these changes, given that now we have a dependency between the more generic coralogix feeder and a spacecat-specific shared-utils package.

HollywoodTonight committed:
feat: integrate logWrapper with automatic trace ID logging

- Add logWrapper from @adobe/spacecat-shared-utils
- Keep using context.log throughout the code (no contextualLog needed)
- The logWrapper now automatically enhances context.log with trace IDs
- No code changes required in consuming services

HollywoodTonight committed:
chore: update to official spacecat-shared-utils@1.76.0

Replace gitpkg temporary dependency with official release version.

Update to @adobe/spacecat-shared-utils@^1.76.0 which includes:
- Enhanced logWrapper with automatic trace ID inclusion
- SQS message trace ID propagation
- HTTP header trace ID support

Related: adobe/spacecat-shared#1154

iuliag commented on package.json:
```diff
@@ -50,6 +50,7 @@
     "@adobe/fetch": "4.2.3",
     "@adobe/helix-shared-wrap": "2.0.2",
     "@adobe/helix-status": "10.1.5",
+    "@adobe/spacecat-shared-utils": "^1.76.1",
```
We use exact versions to keep consistency across the system and clarity of which version might have introduced issues.

HollywoodTonight replied:
Copy that
